### PR TITLE
[teleop] Fix IK unit tests

### PR DESCRIPTION
**Pull request by bli76**
Unit tests were outdated and would fail. Tests were made more lenient and `std::` scope was added where necessary.

### CODE REVIEW

File: jetson/ra_kinematics/arm_state.cpp [hunk 1]
```diff
@@ -55,7 +55,7 @@ ArmState::ArmState(json &geom) : ef_pos_world(Vector3d::Zero()), ef_xform(Matrix
 
     // Sort links by link_num, since they may not be in order from the json
     Link_Comp comparator;
-    sort(collision_avoidance_links.begin(), collision_avoidance_links.end(), comparator);
```
Comment (CameronTressler) [inline jetson/ra_kinematics/arm_state.cpp:58]: yikes good catch

File: jetson/ra_kinematics/meson.build [hunk 2]
```diff
@@ -1,4 +1,4 @@
-project('ra_kinematics', 'cpp', default_options : ['cpp_std=c++1z'])
```
Comment (CameronTressler) [inline jetson/ra_kinematics/meson.build:1]: Is there a reason you changed this? Do we need 14 for the unit tests?

File: jetson/ra_kinematics/test/motion_planner_tests.cpp [hunk 3]
```diff
@@ -47,13 +47,13 @@ TEST(rrt_connect_simple) {
...
 
     target[0] = start[0] + 0.1;
-    target[1] = start[1] + 0.1;
+    target[1] = start[1] - 0.1;
     target[2] = start[2] + 0.1;
     target[3] = start[3] - 0.1;
     target[4] = start[4] - 0.1;
     target[5] = start[5] - 0.1;
 
-    ASSERT_TRUE(solver.is_safe(arm, target));
```
Comment (CameronTressler) [inline jetson/ra_kinematics/test/motion_planner_tests.cpp:56]: Is this failing?...

File: jetson/ra_kinematics/arm_state.cpp (hunk 1 above)
Comment (cgiger00) [reply jetson/ra_kinematics/arm_state.cpp:58]: ^

File: jetson/ra_kinematics/test/motion_planner_tests.cpp (hunk 3 above)
Comment (bli76) [reply jetson/ra_kinematics/test/motion_planner_tests.cpp:56]: I forgot to delete the comment, oops

File: jetson/ra_kinematics/meson.build (hunk 2 above)
Comment (bli76) [reply jetson/ra_kinematics/meson.build:1]: I don't remember changing this... I'll manually change it back

File: jetson/ra_kinematics/test/kinematics_tests.cpp [hunk 4]
```diff
@@ -193,17 +193,16 @@ TEST(ik_test1) {
...
 
     // Create target point vector:
     Vector6d target;
-    target << 0.242980428, -0.0244739898, 0.117660569, -0.426478891, 0.839306191, -1.09910019;
-    //solver.FK(arm);
+    target << 0.26, -0.38, 0.55, 0.3876, 1.209, -0.857;
 
     bool success = false;
-    for (size_t i = 0; i < 10; ++i) {
+    for (size_t i = 0; i < 50; ++i) {
```
[the remark this answers is not in the document]
Comment (bli76) [reply jetson/ra_kinematics/test/kinematics_tests.cpp:205]: It usually takes about 10-20 tries to find a solution in each test from what I've noticed

File: jetson/ra_kinematics/test/config_space_test.cpp [hunk 5]
```diff
@@ -84,7 +84,7 @@ class ConfigSpaceTest {
                 ++i;
             }
 
-            std::pair<Vector6d, bool> ik_solution;
+            std::pair<std::vector<double>, bool> ik_solution;
```
[the remark this answers is not in the document]
Comment (bli76) [reply jetson/ra_kinematics/test/config_space_test.cpp:88]: That happened when we originally moved away from `using namespace std`, we just missed a few lines that needed `std::`.

[the remark this answers is not in the document]
Comment (bli76) [reply jetson/ra_kinematics/test/config_space_test.cpp:88]: yeah pretty much

**Comment by CameronTressler** [comment]
Can we add science arm tests for the motion planning there?

**Review by cgiger00 — APPROVED** [review]
lgtm, had a couple questions about the text versions of the build files but those shouldn't block merging

**Review by catkims — APPROVED** [review]

**Pull request merged**